pkg/banneduser/banneduser_test.go

```diff
@@ -0,0 +1,188 @@
+package banneduser
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	commonsignup "github.com/codeready-toolchain/toolchain-common/pkg/test/usersignup"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestNewBannedUser(t *testing.T) {
+	userSignup1 := commonsignup.NewUserSignup(commonsignup.WithName("johny"), commonsignup.WithEmail("jonhy@example.com"))
+	userSignup1UserEmailHashLabelKey := userSignup1.Labels[toolchainv1alpha1.UserSignupUserEmailHashLabelKey]
+
+	userSignup2 := commonsignup.NewUserSignup(commonsignup.WithName("bob"), commonsignup.WithEmail("bob@example.com"))
+	userSignup2.Labels = map[string]string{}
```

Comment:
```suggestion
```
Let's keep it as `nil`. That would make sure we can handle the case when the map is `nil`.